dev/test/query.ts

```diff
@@ -773,6 +788,209 @@ describe('query interface', () => {
     });
   });
 
+  it('handles stream exception after initialization and heartbeat', () => {
+    const deadlineExceededErr = new GoogleError();
+    deadlineExceededErr.code = Status.DEADLINE_EXCEEDED;
+    deadlineExceededErr.message = 'DEADLINE_EXCEEDED error message';
+
+    let attempts = 0;
+
+    const overrides: ApiOverride = {
+      runQuery: () => {
+        ++attempts;
+
+        // A heartbeat message will initialize the stream, but it is not
+        // a document, so it does not represent progress made on the stream.
+        return stream(heartbeat(1000), deadlineExceededErr);
+      },
+    };
+
+    return createInstance(overrides).then(firestoreInstance => {
+      firestore = firestoreInstance;
+      return firestore
+        .collection('collectionId')
+        .get()
+        .then(() => {
+          throw new Error('Unexpected success in Promise');
+        })
+        .catch(err => {
+          expect(err.message).to.equal('DEADLINE_EXCEEDED error message');
+
+          // The heartbeat initialized the stream before there was a stream
+          // exception, so we only expect a single attempt at streaming.
+          expect(attempts).to.equal(1);
+        });
+    });
+  });
+
+  function handlesRetryableExceptionUntilProgressStops(withHeartbeat: boolean) {
```

Comment:
nit: Can we apply async/await consistently over returning promises. It will also help readability if the function has a return type.

Reply:
I added a return type. I also agree that using async/await is more readable than handling promise results with `then`, `catch`, and `finally`. However, in this case, the test implementation is consistent with other similar tests in this file. I don't think be benefit is worth refactoring all the similar tests.